### PR TITLE
PM 12001 - Fix Empty User Search 500

### PR DESCRIPTION
**Pull request by jrmccannon**
## 🎟️ Tracking

JIRA: https://bitwarden.atlassian.net/browse/PM-12001

## 📔 Objective

When searching for a user that does not exist in the given organization, the serialization fails for sending an empty `IEnumerable<ValueTuple<T1, T2>>`. This is a quick fix to not set the temp data if the enumerable is empty. Also added some null checking to the razor template.

I suspect the DefaultTempDataSerializer has an issue because if an `IEnumerable<ValueTuple<T1, T2>>` with a value in it is returned, it will serialize the data no problem. It only has an issue with an empty enumerable. I did try converting the value over to a `Dictionary<Guid, bool>` and it failed when evaluating during the render. There seems to be a weird interaction between there.

Ideally, we would create a new model to represent the `User` instead of the database model and populate the value on that.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by addisonbeck — DISMISSED** [review]

File: src/Admin/Controllers/UsersController.cs [hunk 1]
```diff
@@ -66,7 +66,9 @@ public async Task<IActionResult> Index(string email, int page = 1, int count = 2
 
         if (_featureService.IsEnabled(FeatureFlagKeys.MembersTwoFAQueryOptimization))
         {
-            TempData["UsersTwoFactorIsEnabled"] = await _twoFactorIsEnabledQuery.TwoFactorIsEnabledAsync(users.Select(u => u.Id));
+            var user2Fa = (await _twoFactorIsEnabledQuery.TwoFactorIsEnabledAsync(users.Select(u => u.Id))).ToList();
+            // TempDataSerializer is having an issue serializing an empty IEnumerable<Tuple<T1,T2>>, do not set if empty.
+            if (user2Fa.Count != 0) TempData["UsersTwoFactorIsEnabled"] = user2Fa;
```
Comment (eliykat) [inline src/Admin/Controllers/UsersController.cs:71]: Code style: please don't use single-line `if` statements.

File: src/Admin/Views/Users/Index.cshtml [hunk 2]
```diff
@@ -72,8 +72,8 @@
                             }
                             @if (featureService.IsEnabled(Bit.Core.FeatureFlagKeys.MembersTwoFAQueryOptimization))
                             {
-                                var usersTwoFactorIsEnabled = TempData["UsersTwoFactorIsEnabled"] as IEnumerable<(Guid userId, bool twoFactorIsEnabled)>;
-                                @if(usersTwoFactorIsEnabled.FirstOrDefault(tuple => tuple.userId == user.Id).twoFactorIsEnabled)
+                                @if(TempData["UsersTwoFactorIsEnabled"] is IEnumerable<(Guid userId, bool twoFactorIsEnabled)> usersTwoFactorIsEnabled
+                                    && usersTwoFactorIsEnabled.FirstOrDefault(tuple => tuple.userId == user.Id).twoFactorIsEnabled)
```
Comment (eliykat) [inline src/Admin/Views/Users/Index.cshtml:76]: If we're using `FirstOrDefault` then we should handle the `null` case (I assume default to `false`). Here we _should_ have a 2FA result for each user in the result set, so I think just `First` would be fine. That said, I haven't tested that locally, and I would be OK if you wanted to take the more cautious approach.

**Review by eliykat — DISMISSED** [review]
Some very minor feedback, but overall looks good to me!

File: src/Admin/Views/Users/Index.cshtml (hunk 2 above)
Comment (jrmccannon) [reply src/Admin/Views/Users/Index.cshtml]: Added a couple checks. The Roslyn analyzer says that the default value of `ValueTuple<Guid, bool>` isn't null, so I broke up the `if` statement into separate variables and used pattern for null checking.

**Review by addisonbeck — APPROVED** [review]

**Pull request merged**